db/queries.py

```diff
@@ -3,44 +3,70 @@
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__icontains=word))
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return Chat.objects.filter(title__startswith="Untitled")
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return list(
+        User.objects.filter(
+            message__sent__year=2015
+        ).values_list("first_name", "last_name")
+    )
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return list(Chat.objects.filter(message__sent__year__gt=2020))
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__icontains=F("user__first_name")))
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return list(
+        User.objects.filter(
+            Q(message__text__istartswith="a")
+            | Q(message__text__istartswith="m"))
+    )
 
 
 def get_delivered_or_admin_messages() -> list[Message]:
-    pass
+    return list(
+        Message.objects.filter(
+            Q(user__username__startswith="admin")
+            | Q(is_delivered__gte=1))
+    )
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return Message.objects.filter(user__first_name=first_name).count()
 
 
 def get_top_users_by_number_of_the_messages() -> list[User]:
-    pass
+    return User.objects.annotate(
+        num_messages=Count("message__sent")
+    ).order_by("-num_messages")[:3]
 
 
 def get_last_5_messages_dicts() -> list[dict]:
-    pass
+    return [
+        {"from": message.user.username, "text": message.text}
+        for message in Message.objects.select_related("user").order_by(
+            "-sent"
+        )[:5]
+    ]
 
 
 def get_chat_dicts() -> list[dict]:
-    pass
+    return [
+        {
+            "id": chat.id,
+            "title": chat.title,
+            "users": [user.username for user in chat.users.all()]
```

Comment:
use values_list() to simplify condition
https://docs.djangoproject.com/en/4.1/ref/models/querysets/#django.db.models.query.QuerySet.values_list

Reply:
i can't figure out how to do it this way

Reply:
list(chat.users.values_list('username', flat=True))
try if this works)